test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala

```diff
@@ -194,6 +194,38 @@ class ConcurrentHashMapTest extends MapTest {
         assertTrue(s"missing value '$value'", encounteredValues.contains(value))
     }
   }
+
+  @Test def testNewKeySet(): Unit = {
+    val keySet = ju.concurrent.ConcurrentHashMap.newKeySet[String]()
+
+    assertFalse(keySet.contains(""))
+
+    assertTrue(keySet.addAll(ju.Arrays.asList("a", "b", "c", "d")))
+    assertFalse(keySet.addAll(ju.Arrays.asList("a", "b", "c", "d")))
+    assertTrue(keySet.contains("a"))
+    assertTrue(keySet.containsAll(ju.Arrays.asList("b", "c")))
+
+    assertTrue(keySet.remove("a"))
+    assertFalse(keySet.contains("a"))
+    assertTrue(keySet.containsAll(ju.Arrays.asList("b", "c", "d")))
+
+    assertTrue(keySet.removeAll(ju.Arrays.asList("b")))
+    assertFalse(keySet.contains("b"))
+    assertTrue(keySet.contains("c"))
+    assertTrue(keySet.contains("d"))
+
+    assertTrue(keySet.retainAll(ju.Arrays.asList("c")))
+    assertTrue(keySet.contains("c"))
+    assertFalse(keySet.contains("d"))
+
+    assertThrows(classOf[UnsupportedOperationException], keySet.contains(null))
+    assertThrows(classOf[UnsupportedOperationException], keySet.containsAll(null))
+    assertThrows(classOf[UnsupportedOperationException], keySet.remove(null))
+    assertThrows(classOf[UnsupportedOperationException], keySet.retainAll(null))
+    assertThrows(classOf[UnsupportedOperationException], keySet.removeAll(null))
+    assertThrows(classOf[UnsupportedOperationException], keySet.add(null))
+    assertThrows(classOf[UnsupportedOperationException], keySet.addAll(null))
```

Comment:
Are you sure these test cases are correct? [`ConcurrentHashMap.containsKey`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#containsKey-java.lang.Object-) should throw an NPE in this case.